### ISSUE TITLE
Fish doesn't print "job stopped" message on ctrl+z

### ISSUE THREAD

**Issue by pvonmoradi** [fish-shell/fish-shell#8158]
Normally, `ctrl+z` sends the current job to background, after which, a message like `[1]+  Stopped                 app_name` should be printed on console.  
This behaviur is inconsistent in fish shell. For example when I do so for `vim`, it behaves correctly (shows `fish: Job 1, 'vim' has stopped`) but when I do it for `v` which is an alias for `vim`, it doesn't print this message. The following screencast shows this:  

https://user-images.githubusercontent.com/1058151/126742316-3ba195c9-0c16-4908-8c24-647332e43fba.mp4

The message `fish: Job 1, “vim $argv” has stopped` is not shown on hitting `ctrl-z` in `v`.

versions:  
`fish, version 3.3.1`
`Linux thinkpad 5.4.0-77-generic #86~18.04.1-Ubuntu`

**Comment by zanchey** [comment]
Yes, functions don't print the job has stopped message, which I can reproduce in 3.0.2 and current git master (c35ffc58f).

**Comment by iacore** [comment]
Should fish functions behave like processes? This is why I don't use fish functions and use fish scripts instead.
